lib/ansible/parsing/mod_args.py

```diff
@@ -332,6 +332,8 @@ def parse(self, skip_action_validation=False):
                         resolved = context.plugin_resolved_collection + '.' + resolved
                     elif AnsibleCollectionRef.is_valid_fqcr(resolved) and resolved.startswith('ansible.legacy.'):
                         resolved = resolved.split('ansible.legacy.')[-1]
+                        if context.plugin_resolved_collection:
```

Comment:
This fixes `ansible.legacy.ping` (which resolves to the builtin plugin) resolving to `ansible.builtin.ping` instead of just 'ping'. I think this makes sense, since `ping` does resolve to `ansible.builtin.ping`.

Reply:
Also simplified this bit when I moved it to PluginLoadContext.